src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedEmbeddedAttributeSymbol : NamedTypeSymbol
+    {
+        private readonly string _name;
+        private readonly NamedTypeSymbol _baseType;
+        private readonly NamespaceSymbol _namespace;
+        private readonly ImmutableArray<Symbol> _members;
+        private readonly ModuleSymbol _module;
+
+        public SynthesizedEmbeddedAttributeSymbol(
+            CSharpCompilation compilation,
+            string @namespace,
+            string name,
+            Func<CSharpCompilation, NamedTypeSymbol, ImmutableArray<Symbol>> getMembers)
+        {
+            _name = name;
+            _baseType = compilation.GetWellKnownType(WellKnownType.System_Attribute);
```

Comment:
>_baseType = compilation.GetWellKnownType(WellKnownType.System_Attribute); [](start = 12, length = 73)

How do we handle use-site diagnostics for this type?

Reply:
Use-site diagnostics are not surfaced in this implementation but are handled by the infrastructure for embedded attributes in `features/readonly-ref`.

Reply:
> Use-site diagnostics are not surfaced in this implementation 

I think we should handle them.

Reply:
And have tests as well.

Reply:
Added.